problems install joplin on macos

**Describe the bug**

I am trying to install `joplin` (cli version) on macos from the unstable channel but I a getting the following error:

```
...
Adding metadata fields to: node_modules/xmlbuilder/package.json
Adding metadata fields to: node_modules/xmlchars/package.json
Adding metadata fields to: node_modules/yallist/package.json
Adding metadata fields to: node_modules/yargs-parser/package.json

> keytar@5.6.0 install /nix/store/a45c2xj0610hracpz9l7dg48kxwsi75r-node_joplin-1.0.168/lib/node_modules/joplin/node_modules/keytar
> prebuild-install || node-gyp rebuild

sh: prebuild-install: command not found
Traceback (most recent call last):
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 50, in <module>
    sys.exit(gyp.script_main())
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 554, in script_main
    return main(sys.argv[1:])
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 547, in main
    return gyp_main(args)
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 532, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 2215, in GenerateOutput
    part_of_all=qualified_target in needed_targets)
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 823, in Write
    self.Pchify))
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 1218, in WriteSources
    cflags = self.xcode_settings.GetCflags(configname)
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 483, in GetCflags
    sdk_root = self._SdkPath()
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 452, in _SdkPath
    return self._XcodeSdkPath(sdk_root)
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 456, in _XcodeSdkPath
    sdk_path = self._GetSdkVersionInfoItem(sdk_root, 'Path')
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 439, in _GetSdkVersionInfoItem
    return GetStdoutQuiet(['xcodebuild', '-version', '-sdk', sdk, infoitem])
  File "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1323, in GetStdoutQuiet
    job = subprocess.Popen(cmdlist, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/nix/store/xifgh5qz2pvafhxcfvldgpa44yk7mppw-python-2.7.18/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/nix/store/xifgh5qz2pvafhxcfvldgpa44yk7mppw-python-2.7.18/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/bin/node" "/nix/store/5q6m7rfs1wqbgsycy75dd25z40yimwhz-nodejs-12.18.4/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /nix/store/a45c2xj0610hracpz9l7dg48kxwsi75r-node_joplin-1.0.168/lib/node_modules/joplin/node_modules/keytar
gyp ERR! node -v v12.18.4
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! keytar@5.6.0 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the keytar@5.6.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /private/var/folders/wt/hv6kvypd65vg0fl9nttvj9440000gn/T/nix-build-node_joplin-1.0.168.drv-0/.npm/_logs/2020-09-23T07_36_39_494Z-debug.log

builder for '/nix/store/ki156zb2ddri9qfzqaa1apz804w60gyr-node_joplin-1.0.168.drv' failed with exit code 1
error: build of '/nix/store/ki156zb2ddri9qfzqaa1apz804w60gyr-node_joplin-1.0.168.drv' failed
```

**To Reproduce**

Steps to reproduce the behavior:

1. Run `nix-env -iA nixpkgs.joplin`

**Expected behavior**
joplin should be installed correctly

**Screenshots**
N/A

**Additional context**


**Notify maintainers**
Unfortunaltely maintainers are not specified :(

**Metadata**

- system: `"x86_64-darwin"`
- host os: `Darwin 19.6.0, macOS 10.15.6`
- multi-user?: `no`
- sandbox: `no`
- version: `nix-env (Nix) 2.3.7`
- channels(dbarrosop): `"darwin, home-manager, nixpkgs-21.03pre243895.72b9660dc18"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

--- a/pkgs/development/node-packages/default.nix
+++ b/pkgs/development/node-packages/default.nix
@@ -285,6 +285,9 @@ let
         libsecret
         self.node-gyp-build
         self.node-pre-gyp
+      ] ++ lib.optionals stdenv.isDarwin [
+        darwin.apple_sdk.frameworks.AppKit
+        darwin.apple_sdk.frameworks.Security
       ];
     };
 

